test_optimizations.py

```
@@ -0,0 +1,264 @@
+from crawler_mcp.core.connection_pool import get_pool
+from crawler_mcp.core.caching import get_embedding_cache, get_query_cache
+from crawler_mcp.core.resilience import get_circuit_breaker
+from crawler_mcp.config import settings
+
+
+async def test_imports():
+    """Test that all imports work correctly."""
+    print("✓ Testing imports...")
+    
+    # Test vector service (modular)
+    try:
```

Ruff (0.12.2)

`crawler_mcp.core.vectors.VectorService` imported but unused; consider using `importlib.util.find_spec` to test for availability

(F401)